### PR TITLE
feat(core-flows,framework,medusa): list shipping options pass in cart as pricing context

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds cart as pricing context to shipping options get route
- renames cart_total to total to align with cart context
- refactors listShippingOptions workflow and route

RESOLVES CMRC-751

### CODE REVIEW

File: packages/medusa/src/api/store/shipping-options/route.ts [hunk 1]
```diff
@@ -1,52 +1,17 @@
...
+import { HttpTypes } from "@medusajs/framework/types"
 
 export const GET = async (
-  req: MedusaRequest<HttpTypes.StoreGetShippingOptionList>,
+  req: MedusaRequest<{}, HttpTypes.StoreGetShippingOptionList>,
   res: MedusaResponse<HttpTypes.StoreShippingOptionListResponse>
 ) => {
-  const { cart_id, is_return } =
-    req.filterableFields as StoreGetShippingOptionsType
+  const { cart_id, is_return } = req.filterableFields
```
Comment (olivermrbl) [inline packages/medusa/src/api/store/shipping-options/route.ts:9]: 🧹 👍

File: packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts [hunk 2]
```diff
@@ -15,26 +16,52 @@ export const listShippingOptionsForCartWorkflowId =
...
+        "shipping_address.country_code",
+        "shipping_address.province",
+        "total",
+      ],
+      options: { throwIfKeyNotFound: true },
+    }).config({ name: "get-cart" })
+
+    const cart = transform({ cartQuery }, ({ cartQuery }) => cartQuery.data[0])
+
+    validatePresenceOfStep({
```
Comment (olivermrbl) [inline packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts:38]: comment: At some point, we could have a more generic validation step that integrates with zod for more sophisticated validation

File: packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts [hunk 3]
```diff
@@ -43,6 +70,36 @@ export const listShippingOptionsForCartWorkflow = createWorkflow(
       }
     )
 
+    const customerGroupIds = when({ cart }, ({ cart }) => {
+      return !!cart.id
```
Comment (olivermrbl) [inline packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts:74]: thought: Wouldn't the workflow have thrown earlier if there were no cart?

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts (hunk 3 above)
Comment (riqwan) [reply packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts:74]: good catch, this is supposed to be `cart.customer_id`

File: packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts (hunk 2 above)
Comment (riqwan) [reply packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts:38]: oh, thats a cool idea!

File: packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts (hunk 3 above)
Comment (riqwan) [reply packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts:74]: will add a test case for both scenarios

Comment (riqwan) [reply packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts:74]: bringing this into the new PR

**Pull request merged**